### PR TITLE
[#2127] React hooks approach on Dataset view

### PR DESCRIPTION
**Pull request by tangrammer**
- [ ] **Update release notes if necessary**

Hooks approach are better to use so let's take this client RQG work as an opportunity

Also work on https://github.com/akvo/akvo-lumen/issues/2712 by https://github.com/akvo/akvo-lumen/commit/c80193cdfa95c969bcfb6d251d9fc05588210a61

### CODE REVIEW

File: client/src/components/common/PendingSaving.jsx [hunk 1]
```diff
@@ -0,0 +1,65 @@
+import { useState, useEffect, useRef } from 'react';
+import PropTypes from 'prop-types';
+import { SAVE_COUNTDOWN_INTERVAL, SAVE_INITIAL_TIMEOUT } from '../../constants/time';
+
+export default function PendingSaving(props) {
```
Comment (marvinkome) [inline client/src/components/common/PendingSaving.jsx:5]: Is `PendingSaving` a hook or a component? It looks like a component but it doesn't return any jsx?

Comment (tangrammer) [reply client/src/components/common/PendingSaving.jsx:5]: it's a hook designed to be composed

any ideas to improve the syntax so it looks like a hook?

Comment (marvinkome) [reply client/src/components/common/PendingSaving.jsx:5]: To make it a hook just rename it to `usePendingSaving`

Comment (tangrammer) [reply client/src/components/common/PendingSaving.jsx]: done :)

**Pull request merged**